### PR TITLE
gradle: 7.3.3 -> 7.4

### LINKED ISSUE #158591: Gradle 7.4
Gradle 7.4 is out, see https://github.com/gradle/gradle/releases/tag/v7.4.0

(I am one of the maintainers, this is a TODO.)

### PR DESCRIPTION
**Pull request by lorenzleutgeb**
Resolves #158591

### CODE REVIEW

File: pkgs/development/tools/build-managers/gradle/update.sh [hunk 1]
```diff
@@ -35,22 +35,32 @@ do
 
     f="gradle-${v}-spec.nix"
 
-    if [ -f "$f" ]
+    if [ "$1" != "" ] && [ "$1" != "$v" ]
```
Comment (kevincox) [inline pkgs/development/tools/build-managers/gradle/update.sh:38]:
```suggestion
    if [[ -n "$1" && "$1" != "$v" ]]
```

**Review by kevincox — DISMISSED** [review]

**Comment by kevincox** [comment]
Result of `nixpkgs-review pr 159226` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>diffoscope</li>
    <li>gscan2pdf</li>
    <li>key</li>
    <li>pdfchain</li>
    <li>pdftk</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>apkleaks</li>
    <li>ghidra</li>
    <li>gradle</li>
    <li>jadx</li>
    <li>kotlin-language-server</li>
    <li>scenic-view</li>
  </ul>
</details>

**Comment by kevincox** [comment]
All of the failed packages worked on the previous commit. Can you please investigate?

```
% nix build -f . diffoscope gscan2pdf key pdfchain pdftk
0 ~/p/nixpkgs#nixpkgs-review/1~2
8:06:35% l result*
lrwxrwxrwx 1 kevincox users 58 2022-02-11 08:06 result -> /nix/store/1d41m6qjqfxra5nywqsd8gy9l4w1kkph-diffoscope-203
lrwxrwxrwx 1 kevincox users 71 2022-02-11 08:06 result-1 -> /nix/store/w3zl97rvir2z5zb7gzkn03n7rqj0g0rp-perl5.34.0-gscan2pdf-2.12.4
lrwxrwxrwx 1 kevincox users 54 2022-02-11 08:06 result-2 -> /nix/store/kyh0p7qd2gii4avfffnmx64qzmfgcj7v-key-2.10.0
lrwxrwxrwx 1 kevincox users 60 2022-02-11 08:06 result-3 -> /nix/store/2c62zsyp82kiq8wx9wd7h3d9vcdjsyly-pdfchain-0.4.4.2
lrwxrwxrwx 1 kevincox users 55 2022-02-11 08:06 result-4 -> /nix/store/62v9r1fmm1h7pvgqwfj645phd6fszds7-pdftk-3.3.2
```

**Review by liff — APPROVED** [review]

**Comment by lorenzleutgeb** [comment]
The dependents should now build again. Is there a way to automate this process, or tell me which dependents break when I update Gradle? It's annoying for me, and very probably also for my reviewers...

**Comment by liff** [comment]
The PR template has this item:

> Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)

That command should do what you need, I think?

**Comment by lorenzleutgeb** [comment]
> The PR template has this item:
> 
> > Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
> 
> That command should do what you need, I think?

Right. I was fantasizing about a solution that would automatically update the hashes in dependents. I should better take a closer look at `nixpkgs-review`. Sorry.

File: pkgs/applications/science/logic/key/default.nix [hunk 2]
```diff
@@ -38,7 +38,8 @@ let
         | sh
     '';
     outputHashMode = "recursive";
-    outputHash = "sha256-1TWySkS8w7L6Q+V946kcLOnM4hL3fieFvLrF5BZAlh4=";
+    outputHashAlgo = "sha256";
+    outputHash = "sha256-GjBUwJxeyJA6vGrPQVtNpcHb4CJlNlY4kHt1PT21xjo=";
```
Comment (fgaz) [inline pkgs/applications/science/logic/key/default.nix:42]: isn't "sha256" already implied by the sri hash?

**Comment by kevincox** [comment]
Result of `nixpkgs-review pr 159226` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>apkleaks</li>
    <li>diffoscope</li>
    <li>ghidra</li>
    <li>gradle (gradle_7)</li>
    <li>gscan2pdf</li>
    <li>jadx</li>
    <li>key</li>
    <li>kotlin-language-server</li>
    <li>pdfchain</li>
    <li>pdftk</li>
    <li>scenic-view</li>
  </ul>
</details>

**Review by kevincox — APPROVED** [review]

File: pkgs/applications/science/logic/key/default.nix (hunk 2 above)
Comment (kevincox) [reply pkgs/applications/science/logic/key/default.nix:42]: Seems reasonable to me. I'll merge and we can always take alternative approaches in a follow-up.

Comment (fgaz) [reply pkgs/applications/science/logic/key/default.nix:42]: > Is there a quicker way to do this?

I usually change a single character in the hash or replace it with `lib.fakeHash`

Comment (lorenzleutgeb) [reply pkgs/applications/science/logic/key/default.nix:42]: Thanks for the tip. I'll use `lib.fakeHash` in the future.

**Pull request merged**